### PR TITLE
Draft of design outline

### PR DESCRIPTION
**Pull request by AutumnVulpes**

### CODE REVIEW

**Comment by methylDragon** [comment]
Could you provide a description

**Review by methylDragon — CHANGES_REQUESTED** [review]
This format is hard to parse, I don't understand what the bullets are for since there are no sections and no descriptions of those sections.

Please use the appropriate headers, making sure to cover:

- What are you trying to do?
- Why are you doing it?
- Why is the method you picked better than the other alternatives (and what are those alternatives?)

**Comment by AutumnVulpes** [comment]
Updated design doc, reverted back to draft.
Completed: Introduction, Context and Design flow.
WIP: Alternatives considered
